packages/x-components/src/x-modules/url/components/url-handler.vue

```diff
@@ -255,10 +256,13 @@
           const urlKey = this.getUrlKey(name);
           if (urlSearchParams.has(urlKey)) {
             if (name in initialUrlState) {
-              const urlValue = urlSearchParams.getAll(urlKey);
+              let urlValue = urlSearchParams.getAll(urlKey);
+              urlValue[0] = sanitize(urlValue[0]);
```

Comment:
To be safe we should sanitize all values and not just the first one. The query only has one occurrence but potentially any parameter could be malicious.

Reply:
`urlValue` is not only `query` param, all URL params are passing one by one. `[0]` is necessary because urlValue is an array of only one element, the param that is being checked.

For example, try to search for a query and select a filter, then write the special characters (`<>`) in the filter and in the query.

Reply:
Sorry,  I think I did not explain properly 😅 What I meant is that a url param can have more than one occurrence. Think about filters, you can select more than one filter at the same time. In that scenario you get an array with multiple entries like:

`['categoryIds:12fad53d7', 'price:1600.0-2000.0']` 

so if you only pick the first element, you will fail to sanitize the other values.